### PR TITLE
MCH: Added Processing of Digits and Preclusters

### PR DESCRIPTION
**Pull request by seperrin**
- Added PhysicsTasks for digits and preclusters processing
- Added basic Check of occupancy level
- Added Reductor and PostProcessing code for TrendingTasks

### CODE REVIEW

**Review by Barthelemy — COMMENTED** [review]
Thank you for your contribution, great job !
I don't have many comments. Once you have considered the one about the PostProcessing, let me know and I will approve and merge.

File: Modules/MUON/MCH/include/MCH/MCHPostProcessing.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
```
[the remark this answers is not in the document]
Comment (seperrin) [reply Modules/MUON/MCH/include/MCH/MCHPostProcessing.h:1]: I think I call it at some point in a .json config file for the TrendingTask, I must have thought that it was useful to have a copy of TrendingTask in the MCH repo but now I don't understand why... I'll try and see what happens if I use TrendingTask in my json instead, and correct all this.

File: Modules/MUON/MCH/src/Decoding.cxx [hunk 2]
```diff
@@ -465,7 +470,7 @@ decode_state_t Add1BitOfData(uint32_t gbtdata, DualSampa& dsr, DualSampaGroup* d
                   (unsigned long)ds->header.fBunchCrossingCounter, (int)ds->header.fPayloadParity);
         int parity = CheckDataParity(ds->data);
         if (parity)
-          fprintf(flog, "===> SAMPA [%2d]: WARNING Parity %d\n", ds->id, parity);
+          fprintf(flog, "===> SAMPA [%d %d %2d]: WARNING Parity %d\n", ds->hit.cru_id, ds->hit.link_id, ds->id, parity);
```
[the remark this answers is not in the document]
Comment (seperrin) [reply Modules/MUON/MCH/src/Decoding.cxx:473]: @aferrero2707 , do we still need the file log ?

File: Modules/MUON/MCH/include/MCH/MCHPostProcessing.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (seperrin) [reply Modules/MUON/MCH/include/MCH/MCHPostProcessing.h:1]: Yep, it works perfectly fine without this file, I'll remove it then and use TrendingTask instead :)

File: Modules/MUON/MCH/src/Decoding.cxx (hunk 2 above)
[the remark this answers is not in the document]
Comment (aferrero2707) [reply Modules/MUON/MCH/src/Decoding.cxx:473]: @seperrin @Barthelemy we can safely remove those fprintf statements. In fact, the whole internal decoding will become obsolete once we will have moved all our QC tasks to the official O2 decoding. We will only keep it around for a short while in order to be able to do some crosschecks...

**Comment by Barthelemy** [comment]
@awegrzyn I get an error on macOS builder with the compilation of Monitoring. Does it ring a bell ?

**Comment by awegrzyn** [comment]
Nope, I've just tried on my Mac and no error :|

**Comment by awegrzyn** [comment]
@Barthelemy it looks like that now it fails on QC creating Configuration.

**Comment by Barthelemy** [comment]
@awegrzyn ok, on it

**Review by Barthelemy — APPROVED** [review]

**Pull request merged**